dts/bindings/spi/nxp,imx-flexspi.yaml

```diff
@@ -13,3 +13,50 @@ properties:
 
     interrupts:
       required: true
+
+    ahb-base:
+      type: int
+      required: true
+      description: |
+        AHB address for read/write access to serial flash memory.
+
+    ahb-bufferable:
+      type: boolean
+      required: false
+      description: |
+        Enable AHB bufferable write access.
+
+    ahb-cacheable:
+      type: boolean
+      required: false
+      description: |
+        Enable AHB cacheable read access.
+
+    ahb-prefetch:
+      type: boolean
+      required: false
+      description: |
+        Enable AHB read prefetch.
+
+    ahb-read-addr-opt:
+      type: boolean
+      required: false
+      description: |
+        AHB read address option bit.
+
+    combination-mode:
+      type: boolean
+      required: false
+      description: |
+        Combine port A and port B data pins to support octal mode access.
+
+    xip:
```

Comment:
What do we need this vs CONFIG_XIP?

Reply:
So we can tell which flash we're using for XIP

Reply:
I seems like we should in theory, but would need a bit more of a use case as your question implies a scenario with multiple flash's.

I assume in the case we'd be able to tell if CONFIG_XIP + looking at `zephyr,flash-controller` or looking if `zephyr,flash` is inside the MMIO region for the specific controller instance.

Reply:
The `mimxrt1064_evk` board is a scenario with multiple flashes - one internal to the SoC package connected to the FLEXSPI2 controller, and a second one mounted on the board connected to the FLEXSPI1 controller. I introduced this property to avoid reinitializing FLEXSPI2 while we're trying to execute from the internal flash (it was already initialized by the boot ROM); otherwise code execution goes off in the weeds.

As @pabigot noted, an alternative approach would be to set `status = "disabled"` on FLEXSPI2, however we are using that node to generate `FLASH_SIZE` and `FLASH_BASE_ADDRESS`. It's effectively the `zephyr,flash` chosen node.

Reply:
So looks like we can make flexspi2 as `status="disabled"`.

So lets go with that path and remove the `xip` property.